### PR TITLE
sync shelve

### PR DESCRIPTION
**Pull request by sloev**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_hubspot/hubspot.py [hunk 1]
```diff
@@ -402,6 +402,7 @@ def store_ids_submissions(self, record: Dict):
             for form_time in forms_times:
                 guid = form_time.split(":", 1)[0]
                 self.event_state["hs_calculated_form_submissions_guids"][guid] = None
+                self.event_state["hs_calculated_form_submissions_guids"].sync()
```
Comment (paddie) [inline tap_hubspot/hubspot.py]: do we want to do it after the loop or at certain intervals instead of every time?

File: tap_hubspot/hubspot.py [hunk 2]
```diff
@@ -421,6 +422,7 @@ def store_ids_submissions(self, record: Dict):
             # we use it to deduplicate and later to iterate
             # we dont care about the value only the key
             self.event_state["contacts_events_ids"][contact_id] = None
+            self.event_state["contacts_events_ids"].sync()
```
Comment (paddie) [inline tap_hubspot/hubspot.py:425]: same as above - otherwise, it makes sense

**Review by paddie — APPROVED** [review]

File: tap_hubspot/hubspot.py [hunk 3]
```diff
@@ -402,7 +402,7 @@ def store_ids_submissions(self, record: Dict):
             for form_time in forms_times:
                 guid = form_time.split(":", 1)[0]
                 self.event_state["hs_calculated_form_submissions_guids"][guid] = None
-                self.event_state["hs_calculated_form_submissions_guids"].sync()
+            self.event_state["hs_calculated_form_submissions_guids"].sync()
```
Comment (paddie) [inline tap_hubspot/hubspot.py:405]: not the other one?

**Review by paddie — APPROVED** [review]

File: tap_hubspot/hubspot.py (hunk 3 above)
Comment (sloev) [reply tap_hubspot/hubspot.py:405]: the other one is singular

**Review by paddie — APPROVED** [review]

**Pull request merged**